docs/reference/text-structure/apis/test-grok-pattern.asciidoc

```diff
@@ -0,0 +1,97 @@
+[role="xpack"]
+[[test-grok-pattern]]
+= Test Grok pattern API
+
+++++
+<titleabbrev>Test Grok pattern</titleabbrev>
+++++
+
+Tests a Grok pattern on lines of text, see also <<grok,Grokking grok>>.
+
+[[test-grok-pattern-request]]
+== {api-request-title}
+
+`GET _text_structure/test_grok_pattern` +
+
+`POST _text_structure/test_grok_pattern` +
+
+[[test-grok-pattern-desc]]
+== {api-description-title}
+
+The test Grok pattern API allows you to execute a Grok pattern on one
+or more lines of text. It returns whether the lines match the pattern
+together with the offsets and lengths of the matched substrings.
+
+[[test-grok-pattern-parms]]
+
+[[test-grok-pattern-query-parms]]
+== {api-query-parms-title}
+
+`ecs_compatibility`::
+(Optional, string) The mode of compatibility with ECS compliant Grok patterns.
+Use this parameter to specify whether to use ECS Grok patterns instead of
+legacy ones when the structure finder creates a Grok pattern. Valid values
+are `disabled` and `v1`. The default value is `disabled`. This setting primarily
+has an impact when a whole message Grok pattern such as `%{CATALINALOG}`
+matches the input. If the structure finder identifies a common structure but
+has no idea of meaning then generic field names such as `path`, `ipaddress`,
+`field1` and `field2` are used in the `grok_pattern` output, with the intention
+that a user who knows the meanings rename these fields before using it.
```

Comment:
```suggestion
matches the input.
```

The last sentence relates to the find structure API and doesn't really make sense for this endpoint.

Reply:
Oops. sorry about that. Should be a little bit more careful with copy-pasting